app_feup/test/integration/src/exams_page_test.dart

```diff
@@ -82,9 +79,83 @@ void main() {
       await completer.future;
 
       await tester.pumpAndSettle();
+      expect(find.byKey(Key(sdisExam.toString())), findsOneWidget);
+      expect(find.byKey(Key(sopeExam.toString())), findsOneWidget);
+    });
 
+    testWidgets('Filtered Exams', (WidgetTester tester) async {
+      final store = Store<AppState>(appReducers,
+          initialState: AppState({
+            'session': Session(authenticated: true),
+            'currUcs': [sopeCourseUnit, sdisCourseUnit],
+            'exams': List<Exam>(),
+            'profile': profile,
+            'filteredExams': filteredExams
+          }),
+          middleware: [generalMiddleware]);
+
+      NetworkRouter.httpClient = mockClient;
+
+      final mockHtml =
+          File('integration/resources/exam_example.html').readAsStringSync();
+      when(mockResponse.body).thenReturn(mockHtml);
+      when(mockResponse.statusCode).thenReturn(200);
+      when(mockClient.get(any, headers: anyNamed('headers')))
+          .thenAnswer((_) async => mockResponse);
+
+      final Completer<Null> completer = Completer();
+      final actionCreator =
+          getUserExams(completer, ParserExams(), Tuple2('', ''));
+
+      final widget = testableReduxWidget(child: ExamsPageView(), store: store);
+
+      await tester.pumpWidget(widget);
+
+      expect(find.byKey(Key(sdisExam.toString())), findsNothing);
+      expect(find.byKey(Key(sopeExam.toString())), findsNothing);
+
+      actionCreator(store);
+
+      await completer.future;
+
+      await tester.pumpAndSettle();
       expect(find.byKey(Key(sdisExam.toString())), findsOneWidget);
       expect(find.byKey(Key(sopeExam.toString())), findsOneWidget);
+
+      final filterIcon = find.byIcon(Icons.settings);
+      expect(filterIcon, findsOneWidget);
+
+      await tester.pumpAndSettle();
+      final IconButton filterButton = find
+          .widgetWithIcon(IconButton, Icons.settings)
+          .evaluate()
+          .first
+          .widget;
+      filterButton.onPressed();
+      await tester.pumpAndSettle();
+
+      expect(find.byType(AlertDialog), findsOneWidget);
+      expect(find.byType(Checkbox), findsNWidgets(Exam.getExamTypes().length));
+
+      final Checkbox mtCheckbox =
+          find.byKey(Key('ExamCheck' + 'Mini-testes')).evaluate().first.widget;
+
+      expect(find.byWidget(mtCheckbox), findsOneWidget);
+      expect(mtCheckbox.value, true);
+      await tester.tap(find.byWidget(mtCheckbox));
+      await completer.future;
+
+      await tester.pumpAndSettle();
+
+      final FlatButton okButton =
+          find.widgetWithText(FlatButton, 'Confirmar').evaluate().first.widget;
+      expect(find.byWidget(okButton), findsOneWidget);
+
+      okButton.onPressed();
+      await tester.pumpAndSettle();
+
+      expect(find.byKey(Key(sdisExam.toString())), findsNothing);
+      expect(find.byKey(Key(sopeExam.toString())), findsNothing);
     });
```

Comment:
Can we create another test case for this whenever the filters want to display a type of exam that does not exist? I'm scared this could cause unexpected behavior whenever an exam type disappears and would love to have this  covered by the tests. Wdyt?

Reply:
This test does not make a lot of sense with this implementation since all the checkboxes are initially checked.
Despite of this, I have created an ExamDoesNotExist test,  in which, even if the Exam-filter map has an unknown type, only the checkboxes that are already know appear.

Reply:
Alright, cool!